### DISCUSSION TITLE
Is it possible to inject code to Sandpack?

### DISCUSSION THREAD

**Discussion by gyurisc** [codesandbox/sandpack#916] [Q&A]
I was wondering if it is possible to inject code dynamically into the Sandpack component. 
I would like to use the SandpackPreview and CodeViewer but I would like to control to code entered from my own editor. 

My use-case is that while the user types the input in my editor control the Sandpack component is updated dynamically.
Is this possible to achieve?

**Comment by danilowoz** [comment]
Hey, sorry for the late answer. Can you provide a bit more information? Maybe a diagram can help.

If I understood correctly, I'd suggest you to use your own code-editor, which you can interact with and pass custom values, and then you can update the file content on Sandpack via `sandpack.updateFIile` whenever you want. So, you would end up with a fake editor where is outside of Sandpack land, but still you can interact with Sandpack preview. Does it make sense?
